erpnext/erpnext_integrations/doctype/shopify_settings/sync_product.py

```diff
@@ -95,7 +95,7 @@ def create_item(shopify_item, warehouse, has_variant=0, attributes=None,variant_
 		"variant_of": variant_of,
 		"sync_with_shopify": 1,
 		"is_stock_item": 1,
-		"item_code": cstr(shopify_item.get("item_code")) or cstr(shopify_item.get("id")),
+		"item_code": shopify_item.get("sku") or get_sku(shopify_item) or cstr(shopify_item.get("item_code")),
```

Comment:
This would break consistency for existing users.

If you instead want incoming Shopify ID to be mapped to an existing item if Shopify SKU == ERPNext ITEM code. Then better way would be to match them before creating the item. 

You could instead try and modify this function to match with an existing item and update it instead of creating new item.

https://github.com/frappe/erpnext/blob/b0e77cbd89394a609108768466fef97cb00902bf/erpnext/erpnext_integrations/doctype/shopify_settings/sync_product.py#L258